### PR TITLE
Simplify and modernize password generation logic

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

Drops legacy fallbacks for password generation and simplifies the code we use.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/21410

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/utils/functions.ts [hunk 1]
```diff
@@ -119,14 +119,14 @@ function capitalize(str: string): string {
 
 function generatePassword(): string {
   const length = Math.floor(Math.random() * 10) + 15;
+  // See https://stackoverflow.com/a/68617567/1470607
   const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789:-_";
+  const array = new Uint32Array(charset.length);
+  window.crypto.getRandomValues(array);
+
   let password = "";
-  if (window.crypto && window.crypto.getRandomValues) {
```
Comment (parlt91) [inline web/html/src/utils/functions.ts:124]: Can we ensure that `window.crypto` is alsways defined or should we keep the check?
I'd guess the answer is yes but it isn't immediately obvious to me.

File: web/html/src/utils/functions.ts [hunk 2]
```diff
@@ -119,14 +119,14 @@ function capitalize(str: string): string {
...
-    var rand = new Uint16Array(length);
-    window.crypto.getRandomValues(rand);
-    for (let i = 0; i < length; i++) password += charset.charAt(Math.floor((rand[i] * charset.length) / 65536));
-  } else {
-    for (let i = 0; i < length; i++) password += charset.charAt(Math.floor(Math.random() * charset.length));
+  for (let i = 0; i < length; i++) {
+    password += charset[array[i] % charset.length];
```
Comment (parlt91) [inline web/html/src/utils/functions.ts:129]: Is `array[i] % charset.length` always an integer without using `Math.floor`?
Or is there some typescript magic that does that automatically?

**Review by parlt91 — COMMENTED** [review]
Two questions before I give my approval

File: web/html/src/utils/functions.ts (hunk 1 above)
Comment (Etheryte) [reply web/html/src/utils/functions.ts:124]: It's available on all the browsers we support, I think this check was originally added quite a while ago.

File: web/html/src/utils/functions.ts (hunk 2 above)
Comment (Etheryte) [reply web/html/src/utils/functions.ts:129]: We used to index by `Math.random()` in `Math.floor(Math.random() * charset.length)` which would need the floor, now we index by `i` which is from the loop body and will be `0`, `1`, `2` etc. The difference is that now `array` is already full of random values (due to `window.crypto.getRandomValues(array)`) so we can just index into it one element after the other and get random values out.

Comment (parlt91) [reply web/html/src/utils/functions.ts:129]: I was talking about the index in charset[]. As it looked to me it's not always going to be an integer if we divide by `charset.length`

**Review by parlt91 — APPROVED** [review]
LGTM

**Comment by Etheryte** [comment]
Note to self: this needs to be ported to 4.3 once the branch is unlocked.

**Pull request merged**